RELEASE_GUIDE.md

```diff
@@ -10,15 +10,42 @@ We apply a semver-ish versioning scheme, using the following logic:
 
 ## How to release
 
-1. Ensure your shell is in the `master` branch
-2. Ensure you are up-to-date: `git pull upstream master` (where "upstream" references https://github.com/mage/mage)
-3. Determine whether this release is a major, minor or patch release
-4. Run: `npm version major`, `npm version minor` or `npm version patch`. This will have:
+In all the `git` shell commands below, `upstream` references the repository at https://github.com/mage/mage.
+
+1. Ensure your shell is in the `master` branch: `git checkout master`
+2. Ensure you are up-to-date: `git pull upstream master`
+3. Continue below
+
+### A) Minor or Patch release based on master
```

Comment:
I would suggest clearly separating major/minor releases (where we release everything) from patch releases.

Patch releases may happen while we have already merged content meant for the next minor/major, so we would need to cherry-pick in the same way as if we had to cherry-pick for a backport. Sometimes you will want to merge all fixes because you have no undesired commits, but it's probably easier not to think about this edge case for documentation purposes.